pycqed/measurement/detector_functions.py

```diff
@@ -581,6 +596,41 @@ def poll_data(self):
                                 self.UHF_map[UHFname]].nr_sweep_points:
                                 gotem[UHFname][n] = True
             accumulated_time += 0.01 * len(self.UHFs)
+            if print_progress:
+                t_now = time.time()
+                if t_now - t_callback > self.progress_callback_interval:
+                    try:
+                        if self.ro_mode == 'rl':
+                            n_acq = [UHF.qas_0_result_acquired() for UHF in
+                                     self.UHFs]
+                        else:
+                            n_acq = [UHF.qas_0_monitor_acquired() for UHF in
+                                     self.UHFs]
+                        # FIXME: This workaround is needed because the
+                        #  UHF truncates qas_0_result_acquired at 2**18.
+                        #  Moreover, it reports 0 when it is done. In this case,
+                        #  we keep the old value during data transfer,
+                        #  and MC will update the progress after data transfer.
```

Comment:
I would still find it useful to add more clarification to this comment. 

- "in this case": the case is when UHF truncates or when it reports 0? I assume it must be the former but it's a bit confusing to read.
- maybe because I don't understand so well how the UHF works, but here "the old value" refers to n_acq? I.e. it is not a data value but the value of the parameter specifying how many samples have been acquired after the last transfer to PC? What do you mean that we keep this old value during data transfer, because n_acq is increased by n_acq_add?

Maybe the comment is fine and you just need to explain to me what it going on.